src/io/flutter/server/vmService/VMServiceManager.java

```diff
@@ -252,21 +270,25 @@ private void onVmServiceReceived(String streamId, Event event) {
   }
 
   private void maybeAddServiceExtension(String name) {
-    synchronized (pendingServiceExtensions) {
+    synchronized (serviceExtensions) {
```

Comment:
Just curious as to why we need to synchronize on serviceExtensions here vs pendingServiceExtensions?

Reply:
Same as above. The goal is to just simplify the locking so that we are only acquiring 1 lock at a time instead of multiple locks. The existing code appeared alright in practice but I didn't want to have to audit to make sure there was never a competing case where the serviceExtensions lock could be acquired and then the pendingServiceExtensions lock was acquired.

Reply:
Gotcha makes sense. Sorry for the duplicate comment - was having github technical issues.